Cubical/Data/Int/Properties.agda

```diff
@@ -233,6 +205,73 @@ ind-assoc _·_ f g p q base m n (suc o) =
     where
     IH = ind-assoc _·_ f g p q base m n o
 
++-comm : ∀ m n → m + n ≡ n + m
++-comm m (pos n) = ind-comm _+_ pos sucInt refl sucInt+ +sucInt pos0+ m n
++-comm m (negsuc n) = ind-comm _+_ negsuc predInt refl predInt+ +predInt negsuc0+ m n
+
 +-assoc : ∀ m n o → m + (n + o) ≡ (m + n) + o
 +-assoc m n (pos o) = ind-assoc _+_ pos sucInt +sucInt refl (λ _ _ → refl) m n o
 +-assoc m n (negsuc o) = ind-assoc _+_ negsuc predInt +predInt refl +predInt m n o
+
+addEq : ℕ → Int ≡ Int
+addEq zero = refl
+addEq (suc n) = compPath (addEq n) sucPathInt
+
+subEq : ℕ → Int ≡ Int
+subEq n i = addEq n (~ i)
+
+_+'_ : Int → Int → Int
+m +' pos n    = transport (addEq n) m
+m +' negsuc n = transport (subEq (suc n)) m
+
+private
+  Lnegsuc : ∀ n z → z +' negsuc n ≡ z +negsuc n
+  Lnegsuc 0 z = refl
+  Lnegsuc (suc n) z = compPath (Lnegsuc n (predInt z)) (sym (predInt+negsuc n z))
+
+  Lpos : ∀ n z → z +' pos n ≡ z +pos n
+  Lpos zero z = refl
+  Lpos (suc n) z = cong sucInt (Lpos n z)
+  
+  Lemma : ∀ m n → m +' n ≡ m + n
+  Lemma m (pos n) = Lpos n m
+  Lemma m (negsuc n) = Lnegsuc n m
+
++'≡+ : _+'_ ≡ _+_
++'≡+ i m n = Lemma m n i
+
+-- We directly get that addition by a fixed number is an equivalence
+-- without having to do any induction!
+isEquivAddInt' : (m : Int) → isEquiv (λ n → n +' m)
+isEquivAddInt' (pos n)    = isEquivTransport (addEq n)
+isEquivAddInt' (negsuc n) = isEquivTransport (subEq (suc n))
+
+isEquivAddInt : (m : Int) → isEquiv (λ n → n + m)
+isEquivAddInt = subst FamilyOfEquiv +'≡+ isEquivAddInt'
+  where
+  FamilyOfEquiv : (f : Int → Int → Int) → Set
+  FamilyOfEquiv f = (m : Int) → isEquiv (λ n → f n m)
+
+minusPlus : ∀ m n → (n - m) + m ≡ n
```

Comment:
@MatthewDaggitt What is this lemma called in the standard library? I looked in https://github.com/agda/agda-stdlib/blob/master/src/Data/Nat/Properties.agda but I couldn't find it...

Reply:
https://github.com/agda/agda-stdlib/blob/5e8b6aa91adc6d60606e17db7b579be356f72aec/src/Data/Nat/Properties.agda#L1335

It's called `m∸n+n≡m`.

Reply:
Oops, I of course wanted to find it for Int and of course I cannot find it in the Properties file for Int... Is there some way to search for lemmas in Agda using patterns? In `Coq` I would do "Search ((_ - _) + _)" and it would give me all the lemmas of this shape.

Reply:
I think you've actually pulled ahead of us already here! I don't think we have that proof in `Data.Integer.Properties` yet. Not sure about the search functionality. I remember someone did mention that there exists something like that but I've never used it.

Reply:
Ok, thanks! I'm not sure if I like the name "m-n+n=n" (first of all because it doesn't match the statement of the lemma) and second of all because it is not so clear where the parantheses go...